src/backend/plugins/config-kde/config-kde.c

```diff
@@ -0,0 +1,296 @@
+/* config-kde.c
+ *
+ * Copyright 2022-2023 The Libproxy Team
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * SPDX-License-Identifier: LGPL-2.1-or-later
+ */
+
+#include <libpeas/peas.h>
+
+#include "config-kde.h"
+
+#include "px-plugin-config.h"
+#include "px-manager.h"
+
+static void px_config_iface_init (PxConfigInterface *iface);
+G_MODULE_EXPORT void peas_register_types (PeasObjectModule *module);
+
+typedef enum {
+  KDE_PROXY_TYPE_NONE = 0,
+  KDE_PROXY_TYPE_MANUAL,
+  KDE_PROXY_TYPE_PAC,
+  KDE_PROXY_TYPE_WPAD,
+  KDE_PROXY_TYPE_SYSTEM,
+} KdeProxyType;
+
+struct _PxConfigKde {
+  GObject parent_instance;
+
+  char *config_file;
+  gboolean available;
+  GFileMonitor *monitor;
+
+  GStrv no_proxy;
+  char *http_proxy;
+  char *https_proxy;
+  char *ftp_proxy;
+  char *socks_proxy;
+  KdeProxyType proxy_type;
+  char *pac_script;
+};
+
+G_DEFINE_FINAL_TYPE_WITH_CODE (PxConfigKde,
+                               px_config_kde,
+                               G_TYPE_OBJECT,
+                               G_IMPLEMENT_INTERFACE (PX_TYPE_CONFIG, px_config_iface_init))
+
+enum {
+  PROP_0,
+  PROP_CONFIG_OPTION
+};
+
+static void px_config_kde_set_config_file (PxConfigKde *self,
+                                           char        *proxy_file);
+
+static void
+on_file_changed (GFileMonitor      *monitor,
+                 GFile             *file,
+                 GFile             *other_file,
+                 GFileMonitorEvent  event_type,
+                 gpointer           user_data)
+{
+  PxConfigKde *self = PX_CONFIG_KDE (user_data);
+
+  g_debug ("%s: Reloading configuration\n", __FUNCTION__);
+  px_config_kde_set_config_file (self, g_file_get_path (file));
+}
+
+static void
+px_config_kde_set_config_file (PxConfigKde *self,
+                               char        *proxy_file)
+{
+  g_autoptr (GError) error = NULL;
+  g_autofree char *line = NULL;
+  g_autoptr (GFile) file = NULL;
+  g_autoptr (GFileInputStream) istr = NULL;
+  g_autoptr (GDataInputStream) dstr = NULL;
+
+  g_clear_pointer (&self->config_file, g_free);
+  self->config_file = proxy_file ? g_strdup (proxy_file) : g_build_filename (g_get_user_config_dir (), "kioslaverc", NULL);
+  self->available = FALSE;
+
+  file = g_file_new_for_path (self->config_file);
```

Comment:
Is `g_key_file_load_from_file` not compatible?

Reply:
No it is not. :(

Reply:
There is no group tag

Reply:
You mean glib expects the implicit `[General]` group in the file?